src/semi_automatic.rs

```diff
@@ -702,10 +707,18 @@ impl<'a> Fold for ToTupleImplementation<'a> {
 }
 
 /// Extracts the tuple placeholder ident from the given trait implementation.
-fn extract_tuple_placeholder_ident(trait_impl: &ItemImpl) -> Result<Ident> {
+fn extract_tuple_placeholder_ident(trait_impl: &ItemImpl) -> Result<(bool, Ident)> {
+    if let Type::Reference(ref type_ref) = *trait_impl.self_ty {
```

Comment:
Both ifs could be combined into one match. Especially the first one would be only one match arm where have `Type::Reference(Type::Path(ref type_ref)) `

Reply:
I'm happy to update this to a match as follows:
```rust
match *trait_impl.self_ty {
    Type::Reference(ref type_ref) => {
        if let Type::Path(ref type_path) = *type_ref.elem {
            if let Some(ident) = type_path.path.get_ident() {
                return Ok((true, ident.clone()));
            }
        }
    }
    Type::Path(ref type_path) => {
        if let Some(ident) = type_path.path.get_ident() {
            return Ok((false, ident.clone()));
        }
    }
    _ => {}
}
```

However I'm not sure what you mean by using `Type::Reference(Type::Path(ref type_ref))`, `TypeReference` isn't an enum so I'm not sure how to nest the `Type::Path` within the same match statement.

Reply:
Ahh yeah that is fine, I had overseen the `type_ref.elem`. TY